darwin/Platform.c

```diff
@@ -120,21 +120,65 @@ const MeterClass* const Platform_meterTypes[] = {
    NULL
 };
 
-static double Platform_nanosecondsPerMachTick = 1.0;
+static void Platform_getCPUBrandString(char *cpu_brand_string, size_t cpu_brand_string_size) {
+   if (sysctlbyname("machdep.cpu.brand_string", cpu_brand_string, &cpu_brand_string_size, NULL, 0) == -1) {
+      CRT_fatalError("Could not determine what CPU we are running on");
+   }
+}
 
-static double Platform_schedulerTicksPerNS = -1;
+// Adapted from https://developer.apple.com/documentation/apple-silicon/about-the-rosetta-translation-environment
+static bool Platform_isRunningUnderRosetta2() {
```

Comment:
I'd call this instead "Platform_isRunningTranslated" (like the kernel interface name) rather than using the product/marketing name.

Reply:
That's what Apple called their function, but I fear that it's not very clear. It sounds like it has more to do with translation of human languages, not instruction sets.

Reply:
Depends on your background I guess, for me it was immediately clear.  When we're talking kernels and low-level systems code I think most people would understand the term "translated" in the intended way here.

Reply:
Renamed.